tests/unit/test_charm_scaling.py

```diff
@@ -171,3 +171,33 @@ def test_scaling_instance_map_configured(harness: Harness) -> None:
                 "port": 8034,
             },
         }
+
+
+def test_scaling_stream_writers_configured(harness: Harness) -> None:
+    """
+    arrange: charm deployed, integrated with Redis and set as leader.
+    act: scale charm to more than 1 unit.
```

Comment:
Perhaps another test that explicitly checks that stream_writer is not configured when we have only one unit?

Reply:
Nice suggestion! I'll add it to the next PR https://github.com/canonical/synapse-operator/pull/266